### DISCUSSION TITLE
Accessing development server from other device in same network

### DISCUSSION THREAD

**Discussion by hariangr** [inertiajs/inertia#1484] [Help]
I develop a mobile web from a Mac, I want to access the site using my Android device running in the same network

Usually I can access it normally when I host on 0.0.0.0 or using tunnel like ngrok back when I use normal Laravel

Now I want to try Inertiajs, it seems I need to access two running backend in development*, the Node server and the Laravel server. But when I access the Laravel address from my mobile phone, react-refresh throw this error

<img width="1204" alt="image" src="https://user-images.githubusercontent.com/18433715/223914699-2a74764b-90ff-4eb0-8c97-12c6d41a8482.png">

How do I let react-refresh know the IP Address of my development server?

**Comment by hariangr** [comment]
I forgot to mention, when the Node server not running and I already build using `npm run build`, I can access the site normally. I assume this because Laravel use the generated HTML instead of accessing through Node server (This is problematic of course since I need to do npm run dev for every change I want to see on mobile)

**Comment by moham96** [comment]
edit `package.json` dev script from:
```
"dev": "vite"
```
to 
```
"dev": "vite --host 192.168.5.80"
```
replace "192.168.5.80" with the ip of your dev device

**Reply by am05mhz** [reply]
this solution works, but it limits the accessible address specifically to the ip, in which you can access it from other device with the ip, but you also need to access to the same ip even from local machine, and cannot use `localhost` as the address

setting it to 0.0.0.0 wont work, as it makes the browser try to load any resources from 0.0.0.0, not the machine's ip

the answer below by @Lexazan is better

**Comment by Lexazan** [comment]
You may also need to add the following to the vite.config.ts (.js):
```js
export default defineConfig({
//...
    server: {
        host: '0.0.0.0',       // Accept connections from LAN
        port: 5173,           
        strictPort: true,
        hmr: {
            host: '192.168.0.123', // Your machine's local IP address
        }
    },
// ...
});
```
